### PR TITLE
Fix parameter name collision in AutoMLBatchPredictOperator #10723

### LINKED ISSUE #10723: Parameter name collision in AutoMLBatchPredictOperator
Hello,

AutoMLBatchPredictOperator accepts `params` parameter, but it is also used by BaseOperator.
https://github.com/apache/airflow/blob/828f7303b7ba058c3772787529f19162a63ca1a1/airflow/providers/google/cloud/operators/automl.py#L177
https://github.com/apache/airflow/blob/828f7303b7ba058c3772787529f19162a63ca1a1/airflow/models/baseoperator.py#L341
We should think about what we want to do with it.

Best regards,
Kamil

CC: @turbaszek

### PR DESCRIPTION
**Pull request by ghost**
Proposition of solving issue: [#10723 ](https://github.com/apache/airflow/issues/10723)

Fix parameter name collision in `AutoMLBatchPredictOperator` by changing parameter `params` to `prediction_params`.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by turbaszek — APPROVED** [review]
Looks good! @potiuk can you please take a look at the additional info file, I'm wondering if there needs to be something done with backports?

File: airflow/providers/google/cloud/operators/automl.py [hunk 1]
```diff
@@ -300,7 +300,7 @@ def __init__(  # pylint: disable=too-many-arguments
         self.model_id = model_id
         self.location = location
         self.project_id = project_id
-        self.params = params  # type: ignore
+        self.prediction_params = prediction_params  # type: ignore
```
Comment (turbaszek) [inline airflow/providers/google/cloud/operators/automl.py]:
```suggestion
        self.prediction_params = prediction_params
```

File: airflow/providers/google/cloud/operators/automl.py [hunk 2]
```diff
@@ -240,9 +240,9 @@ class AutoMLBatchPredictOperator(BaseOperator):
         written. If a dict is provided, it must be of the same form as the protobuf message
         `google.cloud.automl_v1beta1.types.BatchPredictOutputConfig`
     :type output_config: Union[dict, ~google.cloud.automl_v1beta1.types.BatchPredictOutputConfig]
-    :param params: Additional domain-specific parameters for the predictions, any string must be up to
+    :param prediction_params: Additional domain-specific parameters for the predictions, any string must be up to
```
Comment (turbaszek) [inline airflow/providers/google/cloud/operators/automl.py]:
```suggestion
    :param prediction_params: Additional domain-specific parameters for the predictions, any string must be up to
```
This line is too long

Comment (ghost) [reply airflow/providers/google/cloud/operators/automl.py]: Thank you, last commit shortened these lines

**Review by turbaszek — APPROVED** [review]

**Pull request merged**